### PR TITLE
fix: Update language support

### PR DESCRIPTION
**Pull request by jhildenbiddle**
**Summary**

1. Dynamically set the HTML `lang` attribute based on URL hash. This attribute must be set properly in order to trigger auto-translation features on applicable Chromium-based browsers (Chrome, Edge). Note that this fix works only for the initial page load as `lang` attribute changes are not recognized on hash change. For example:

   https://docsify.js.org/
   
   ```html
   <html lang="en">
   ```

   https://docsify.js.org/#/zh-cn/
   
   ```html
   <html lang="zh-cn">
   ```

2. Remove the `write-a-plugin.md` alias as it appears to be unnecessary
3. Add Russian and Spanish search-related label translations

**What kind of change does this PR introduce?** (check at least one)

- [x] Feature
- [x] Docs

If changing the UI of default theme, please provide the **before/after** screenshot:

* **Before** (notice the lack of the translation option in the URL bar)
  Chrome1

* **After**
  Chrome2

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [x] Safari
- [x] Edge
- [ ] IE

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,99 +1,118 @@
...
+          },
+          paths: 'auto',
+          placeholder: {
+            '/es/': 'Buscar',
+            '/de-de/': 'Suche',
+            '/ru-ru/': 'Поиск',
+            '/zh-cn/': '搜索',
+            '/': 'Search',
+          },
+          // pathNamespaces: ['/es', '/de-de', '/ru-ru', '/zh-cn']
```
Comment (sy-records) [inline index.html:78]:
```suggestion
          pathNamespaces: ['/es', '/de-de', '/ru-ru', '/zh-cn']
```

Let's keep it

**Review by sy-records — APPROVED** [review]

**Comment by sy-records** [comment]
It seems that the test snapshot file needs to be updated

File: index.html (hunk 1 above)
Comment (jhildenbiddle) [reply index.html]: @sy-records Indeed. I meant to uncomment this before I checked it in. Good catch.

**Comment by jhildenbiddle** [comment]
> It seems that the test snapshot file needs to be updated

@sy-records Yep. Good catch. I didn't test locally because I need that changes from #1487 to test on my ARM-based Mac. Should be all set now.

**Review by sy-records — APPROVED** [review]

**Review by Koooooo-7 — APPROVED** [review]
LGTM.

**Pull request merged**